packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -0,0 +1,228 @@
+
+    private renderLeftPanel() {
+        return (
+            <div className={Classes.MULTISTEP_DIALOG_LEFT_PANEL}>
+                {this.getStepChildren().filter(isStepElement).map(this.renderStep)}
+            </div>
+        );
+    }
+
+    private renderStep = (step: StepElement, index: number) => {
+        const stepNumber = index + 1;
+        const hasBeenViewed = this.state.lastViewedIndex >= index;
+        const currentlySelected = this.state.selectedIndex === index;
+        return (
+            <div className={classNames(Classes.STEP_CONTAINER, { [Classes.ACTIVE]: hasBeenViewed })} key={index}>
+                <div
+                    className={classNames(Classes.STEP, { [Classes.ACTIVE]: hasBeenViewed })}
+                    onClick={this.handleClickStep(index)}
+                >
+                    <div className={classNames(Classes.STEP_ICON, { [Classes.ACTIVE]: hasBeenViewed })}>
+                        {stepNumber}
+                    </div>
+                    <div className={classNames(Classes.STEP_TITLE, { [Classes.ACTIVE]: currentlySelected })}>
+                        {step.props.title}
+                    </div>
+                </div>
+            </div>
+        );
+    };
+
+    private handleClickStep = (index: number) => {
+        if (index > this.state.lastViewedIndex) {
+            return undefined;
+        }
+        return this.handleChangeStep(index);
+    };
+
+    private maybeRenderRightPanel() {
+        const steps = this.getStepChildren();
+        if (steps.length <= this.state.selectedIndex) {
+            return null;
+        }
+
+        const stepProp = steps[this.state.selectedIndex].props;
+        if (stepProp.panel === undefined) {
+            return null;
+        }
+
+        const { className, panelClassName } = stepProp;
+        return (
+            <div
+                className={classNames(Classes.MULTISTEP_DIALOG_RIGHT_PANEL, className, panelClassName)}
+                role="steppanel"
+            >
+                {stepProp.panel}
+                {this.renderFooter()}
+            </div>
+        );
+    }
+
+    private renderFooter() {
+        return (
+            <div className={Classes.MULTISTEP_DIALOG_FOOTER}>
+                <div className={Classes.DIALOG_FOOTER_ACTIONS}>{this.renderButtons()}</div>
+            </div>
+        );
+    }
+
+    private renderButtons() {
+        const buttons = [];
+        if (this.state.selectedIndex > 0) {
+            buttons.push(<Button key="back" onClick={this.handleClickBack()} text="Back" />);
+        }
+
+        if (this.state.selectedIndex === this.getStepChildren().length - 1) {
+            buttons.push(this.getFinalButton());
+        } else {
+            buttons.push(
+                <Button
+                    disabled={this.props.disableNext}
+                    intent="primary"
+                    key="next"
+                    onClick={this.handleClickNext()}
+                    text="Next"
```

Comment:
@adidahiya how does BP handle localization? It seems like for most components the client supplies all of the copy. Should that also be the case here (at least making it possible to override the button text)?